idea/idea-fir/src/org/jetbrains/kotlin/idea/fir/intentions/HLImportMemberIntention.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.fir.intentions
+
+import com.intellij.codeInsight.intention.HighPriorityAction
+import org.jetbrains.kotlin.idea.KotlinBundle
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicatorInput
+import org.jetbrains.kotlin.idea.api.applicator.applicator
+import org.jetbrains.kotlin.idea.fir.api.AbstractHLIntention
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicabilityRange
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicatorInputProvider
+import org.jetbrains.kotlin.idea.fir.api.applicator.inputProvider
+import org.jetbrains.kotlin.idea.fir.applicators.ApplicabilityRanges
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.components.ShortenCommand
+import org.jetbrains.kotlin.idea.frontend.api.components.ShortenOption
+import org.jetbrains.kotlin.idea.frontend.api.symbols.*
+import org.jetbrains.kotlin.idea.frontend.api.symbols.markers.KtSymbolKind
+import org.jetbrains.kotlin.idea.frontend.api.symbols.markers.KtSymbolWithKind
+import org.jetbrains.kotlin.idea.references.mainReference
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.psiUtil.getQualifiedElement
+import org.jetbrains.kotlin.psi.psiUtil.isInImportDirective
+
+class HLImportMemberIntention : AbstractHLIntention<KtNameReferenceExpression, HLImportMemberIntention.Input>(
+    KtNameReferenceExpression::class, Companion.applicator
+), HighPriorityAction {
+    override val applicabilityRange: HLApplicabilityRange<KtNameReferenceExpression> get() = ApplicabilityRanges.SELF
+
+    override val inputProvider: HLApplicatorInputProvider<KtNameReferenceExpression, Input> = inputProvider { psi ->
+        val symbol = psi.mainReference.resolveToSymbol() ?: return@inputProvider null
```

Comment:
There are many labeled returns in this function; maybe it would be benefitial to extract the code from `importProvider` to a function?

Reply:
Sounds good!